pkg/apis/nfd/v1alpha1/nodefeaturegroup_types.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeFeatureGroup resource holds Node pools by featureGroup
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,shortName=nfg
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient
+type NodeFeatureGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec NodeFeatureGroupSpec `json:"spec"`
+}
+
+// NodeFeatureGroupSpec describes a NodeFeatureGroup object.
+type NodeFeatureGroupSpec struct {
+	// FeatureGroupRules is a list of feature group rules.
+	// Feature group rules are used to group nodes into feature groups.
+	// +optional
+	FeatureGroup `json:"featureGroup"`
+	// Features is the set of features that are shared by all nodes in the feature group
+	// +optional
+	Features map[string]string `json:"features"`
+}
+
+type FeatureGroup struct {
+	// Name is the name of the feature group
+	Name string `json:"name"`
+	// Nodes is the list of nodes in the cluster that belong to this feature group
+	// +optional
+	Nodes []string `json:"nodes"`
+}
+
+// NodeFeatureGroupList contains a list of NodeFeatureGroup objects.
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type NodeFeatureGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []NodeFeatureGroup `json:"spec"`
+}
+
+// NodeFeatureGroupRule resource specifies a configuration for feature-based
+// node grouping.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,shortName=nfgr
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient
+// +genclient:nonNamespaced
+type NodeFeatureGroupRule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec NodeFeatureGroupRuleSpec `json:"spec"`
+}
+
+// NodeFeatureGroupRuleSpec describes a NodeFeatureGroupRule.
+type NodeFeatureGroupRuleSpec struct {
+	FeatureGroupRules []Rule `json:"featureGroupRules"`
```

Comment:
If we go with the separate CRD, we should specify a separate Rule type for it. Many of the fields (like `labels`, `taints` etc) are not applicable here and should be rejected